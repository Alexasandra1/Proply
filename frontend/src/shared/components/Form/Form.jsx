import React from 'react';
import { Input } from '../Input/Input';
import './Form.scss';

// export function Form({ inputs, formClass = "form", onChange }) {
    export function Form({ inputs, formClass = "form"}) {
    return (
        <form className={formClass}>
            {inputs.map((inputProps, index) => (
                // <Input key={index} {...inputProps} onChange={onChange} />
                <Input key={index} {...inputProps} />
            ))}
        </form>
    );
}
