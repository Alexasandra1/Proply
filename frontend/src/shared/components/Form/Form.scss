.form {
    display: flex;
    flex-direction: column;
    gap: 10px;
    justify-content: center;
}
.registration{
    display: flex;
    flex-wrap: wrap;
    gap: 10px;
    width: 860px;
}


.addHousePage__body__formContainer__container__firstForm{
    display: flex;
    flex-direction: row;
    justify-content: center;
    gap: 2%;
    flex-wrap: wrap;
}
@media (max-width: 740px) {
    .registration{
        width: 400px;
    }
  }
  
  @media (max-width: 340px) {
    .registration{
        width: 300px;
    }
    .form{
        width: 300px;
    }
  }