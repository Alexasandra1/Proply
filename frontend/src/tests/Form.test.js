import React from 'react';
import renderer from 'react-test-renderer';
import { Form } from '../shared/components/Form/Form';

test('renders correctly', () => {
  const inputs = [
    { type: 'text', name: 'test1', value: 'Test 1' },
    { type: 'text', name: 'test2', value: 'Test 2' },
  ];

  const tree = renderer
    .create(<Form inputs={inputs} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
