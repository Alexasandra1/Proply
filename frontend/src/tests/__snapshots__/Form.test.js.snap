// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="form"
>
  <div
    className="input"
  >
    <input
      className="input__input undefined"
      type="text"
      value="Test 1"
    />
  </div>
  <div
    className="input"
  >
    <input
      className="input__input undefined"
      type="text"
      value="Test 2"
    />
  </div>
</form>
`;
